File: src/test/java/MainPage.py
from BasePage import BasePage
from selenium.webdriver.common.by import By

class MainPage(BasePage):
    FEATURED_PRODUCT = (By.CSS_SELECTOR, "#content > div.row .product-thumb")
    PRODUCT_NAME = (By.CSS_SELECTOR, ".description h4 a")
    PRODUCT_PRICE = (By.CSS_SELECTOR, ".description > .price span")
    CAROUSEL_ITEM = (By.CSS_SELECTOR, "#carousel-banner-0")
    HEADER = (By.CSS_SELECTOR, "#top")
    MONEY_LIST = (By.CSS_SELECTOR, "#top > div.row .product-thumb")
    SEARCH_BUTTON = (By.CSS_SELECTOR, "#search button")
    SEARCH_FIELD = (By.XPATH, "//input[@name='search']")
    SEARCH_ICON = (By.XPATH, "//i[@class='fa-solid fa-magnifying-glass']")
    SEARCH_RESULT_HEADER = (By.XPATH, "//*[@id='content']/h1")
    MACBOOK_IMAGE_LINK = (By.XPATH, "//img[@title='MacBook']")
    MACBOOK_INVALID_IMAGE_LINK = (By.XPATH, "//img[@title='MacBook Pro Max']")
    IMAGE_COUNTER = (By.XPATH, "//div[text()='2 of 5']")
    POPUP_CLOSE_BUTTON = (By.XPATH, "/html/body/div[2]/div/button[2]")
    CURRENCY_DROPDOWN = (By.CLASS_NAME, "dropdown-toggle")
    EUR_CURRENCY_LINK = (By.XPATH, "//a[@href='EUR']")
    CURRENCY_SYMBOL = (By.XPATH, "//strong")
    DESKTOPS_LINK = (By.XPATH, f"//a[@href='{BasePage.BASE_URL}/en-gb/catalog/desktops']")
    PC_LINK = (By.XPATH, f"//a[@href='{BasePage.BASE_URL}/en-gb/catalog/desktops/pc']")
    CONTENT_TEXT = (By.ID, "content")

    def __init__(self, driver):
        super().__init__(driver)
        self.driver = driver

    def open_main_page(self):
        self.driver.get(self.BASE_URL)

    def enter_search_text(self, text):
        self.driver.find_element(*self.SEARCH_FIELD).send_keys(text)

    def click_search_button(self):
        self.driver.find_element(*self.SEARCH_ICON).click()

    def get_search_result_header_text(self):
        return self.driver.find_element(*self.SEARCH_RESULT_HEADER).text

    def get_content_text(self):
        return self.driver.find_element(*self.CONTENT_TEXT).text

File: src/test/java/MainPage.java
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;

public class MainPage extends BasePage {
    private static final By FEATURED_PRODUCT = By.cssSelector("#content > div.row .product-thumb");
    private static final By PRODUCT_NAME = By.cssSelector(".description h4 a");
    private static final By PRODUCT_PRICE = By.cssSelector(".description > .price span");
    private static final By CAROUSEL_ITEM = By.cssSelector("#carousel-banner-0");
    private static final By HEADER = By.cssSelector("#top");
    private static final By MONEY_LIST = By.cssSelector("#top > div.row .product-thumb");
    private static final By SEARCH_BUTTON = By.cssSelector("#search button");
    private static final By SEARCH_FIELD = By.xpath("//input[@name='search']");
    private static final By SEARCH_ICON = By.xpath("//i[@class='fa-solid fa-magnifying-glass']");
    private static final By SEARCH_RESULT_HEADER = By.xpath("//*[@id='content']/h1");
    private static final By MACBOOK_IMAGE_LINK = By.xpath("//img[@title='MacBook']");
    private static final By MACBOOK_INVALID_IMAGE_LINK = By.xpath("//img[@title='MacBook Pro Max']");
    private static final By IMAGE_COUNTER = By.xpath("//div[text()='2 of 5']");
    private static final By POPUP_CLOSE_BUTTON = By.xpath("/html/body/div[2]/div/button[2]");
    private static final By CURRENCY_DROPDOWN = By.className("dropdown-toggle");
    private static final By EUR_CURRENCY_LINK = By.xpath("//a[@href='EUR']");
    private static final By CURRENCY_SYMBOL = By.xpath("//strong");
    private static final By DESKTOPS_LINK = By.xpath("//a[@href='" + BasePage.BASE_URL + "/en-gb/catalog/desktops']");
    private static final By PC_LINK = By.xpath("//a[@href='" + BasePage.BASE_URL + "/en-gb/catalog/desktops/pc']");
    private static final By CONTENT_TEXT = By.id("content");

    public MainPage(WebDriver driver) {
        super(driver);
        logger.info("Инициализирована страница");
    }

    public void openMainPage() {
        driver.get(BASE_URL);
        logger.info("открыта страница");
    }

    public void enterSearchText(String text) {
        driver.findElement(SEARCH_FIELD).sendKeys(text);

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}({})", methodName, text);
    }

    public void clickSearchButton() {
        driver.findElement(SEARCH_ICON).click();

        String methodName = Thread.currentThread().getStackTrace()[1].getMethodName();
        logger.info("функция {}", methodName);
    }

    public String getSearchResultHeaderText() {
        return driver.findElement(SEARCH_RESULT_HEADER).getText();
    }

    public String getContentText() {
        return driver.findElement(CONTENT_TEXT).getText();
    }
}
